tools/src/wyvern/tools/parsing/coreparser/arch/ASTComponentTypeDecl.java

```diff
@@ -0,0 +1,175 @@
+                        if (args.size() != reqs.size()) {
+                            // inconsistent dependency errors
+                            ToolError.reportError(
+                                    ErrorMessage.COMPONENT_DEPENDENCY_INCONSISTENCY, this.getLocation(),
+                                    typeName);
+                        }
+                        for (FormalArg f : args) {
+                            String name = f.getName();
+                            String type = ((NominalType) f.getType()).getTypeMember();
+                            if (!reqs.get(name).equals(type)) {
+                                // inconsistent dependency error
+                                ToolError.reportError(
+                                        ErrorMessage.COMPONENT_DEPENDENCY_INCONSISTENCY, this.getLocation(),
+                                        typeName);
+                            }
+                        }
+                    }
+                }
+            }
+
+            /* Check that it exposes the correct fields */
+            List<Declaration> decls = ((New) expr).getDecls().stream()
+                    .filter(p -> p instanceof DefDeclaration)
+                    .collect(Collectors.toList());
+            for (Declaration d : decls) {
+                DefDeclaration dd = (DefDeclaration) d;
+                if (dd.getType() instanceof StructuralType) {
+                    List<DeclType> valdecltypes = ((StructuralType) (dd.getType()))
+                            .getDeclTypes().stream().filter(p -> p instanceof ValDeclType)
+                            .collect(Collectors.toList());
+
+                    for (DeclType dt : valdecltypes) {
+                        ValDeclType vdtype = (ValDeclType) dt;
+                        String name = vdtype.getName();
+                        String t = ((NominalType) vdtype.getSourceType().getValueType())
+                                .getTypeMember();
+                        if (provs.get(name) == null || !provs.get(name).equals(t)) {
+                            // inconsistent fields
+                            ToolError.reportError(
+                                    ErrorMessage.COMPONENT_DEPENDENCY_INCONSISTENCY, this.getLocation(),
+                                    typeName);
+                        } else {
+                            checkCount++;
+                        }
+                    }
+                    if (checkCount != valdecltypes.size()) {
+                        // inconsistent fields
+                        ToolError.reportError(
+                                ErrorMessage.COMPONENT_DEPENDENCY_INCONSISTENCY, this.getLocation(),
+                                typeName);
+                    }
+                }
+            }
+
+        } catch (ToolError e) {
+            e.printStackTrace();
+            return false;
+        }
```

Comment:
I think the try/catch block should be removed. This function should only return a boolean value indicating whether the check passed or failed. The exception should be thrown in DeclCheckVisitor.